internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -238,67 +238,117 @@ func (s Service) ListTargets(ctx context.Context, req *pbs.ListTargetsRequest) (
 	if req.GetPageSize() != 0 && uint(req.GetPageSize()) < s.maxPageSize {
 		pageSize = int(req.GetPageSize())
 	}
-	filter, err := handlers.NewFilter(ctx, req.GetFilter())
-	if err != nil {
-		return nil, err
+	filterItemFn := func(ctx context.Context, item target.Target) (bool, error) {
```

Comment:
these bits might be more simple as as switch with a default for this and a case
for `req.GetFilter() != ""`

Reply:
Sounds good